XiEditor/Search.swift

```diff
@@ -86,139 +249,117 @@ class FindViewController: NSViewController, NSSearchFieldDelegate, NSControlText
         self.view.layer?.backgroundColor = unifiedTitlebar ? newBackgroundColor.cgColor : NSColor.veryLightGray.cgColor
     }
 
-    func control(_ control: NSControl, textView: NSTextView, doCommandBy commandSelector: Selector) -> Bool {
-        switch commandSelector {
-        case #selector(NSResponder.cancelOperation(_:)):
-            // overriding cancelOperation is not enough, because the first Esc would just clear the
-            // search field and not call cancelOperation
-            findDelegate.closeFind()
-            return true
-        default:
-            return false
-        }
+    func showButtons(show: Bool) {
+        addButton.isHidden = !show
+        deleteButton.isHidden = !show
+    }
+
+    func disableAddButton(disable: Bool) {
+        addButton.isEnabled = !disable
+    }
+
+    func disableDeleteButton(disable: Bool) {
+        deleteButton.isEnabled = !disable
+    }
+
+    @IBAction func addSearchField(_ sender: NSButton) {
+        let offset = parentFindView?.view.fittingSize.height
```

Comment:
Small bug here: When multiple queries are present and you paste in something, the edit view gets hidden.

I suspect it's because the scroll view's origin isn't set properly after the paste, or it doesn't account for the presence of the find view. 
<img width="1329" alt="screen shot 2018-09-27 at 09 29 34" src="https://user-images.githubusercontent.com/12481747/46121218-09529200-c23d-11e8-9b9f-3fb56ee22dc9.png">